packages/react/spec/auto/inputs/PolarisAutoHiddenInput.spec.tsx

```diff
@@ -0,0 +1,110 @@
+import { AppProvider } from "@shopify/polaris";
+import translations from "@shopify/polaris/locales/en.json";
+import { act, render } from "@testing-library/react";
+import { userEvent } from "@testing-library/user-event";
+import type { ReactNode } from "react";
+import React from "react";
+import { PolarisAutoForm } from "../../../src/auto/polaris/PolarisAutoForm.js";
+import { PolarisAutoHiddenInput } from "../../../src/auto/polaris/inputs/PolarisAutoHiddenInput.js";
+import { PolarisAutoSubmit } from "../../../src/auto/polaris/submit/PolarisAutoSubmit.js";
+import { testApi as api } from "../../apis.js";
+import { MockClientProvider, mockUrqlClient } from "../../testWrappers.js";
+import { mockWidgetFindBy } from "../support/helper.js";
+import { getWidgetModelMetadata } from "../support/widgetModel.js";
+
+const PolarisMockedProviders = (props: { children: ReactNode }) => {
+  return (
+    <MockClientProvider api={api}>
+      <AppProvider i18n={translations}>{props.children}</AppProvider>
+    </MockClientProvider>
+  );
+};
+
+describe("PolarisAutoHiddenInput", () => {
+  it("should set the value", async () => {
+    const user = userEvent.setup();
+
+    const { getByRole } = render(
+      <PolarisAutoForm action={api.widget.create}>
+        <PolarisAutoHiddenInput field="name" value="Bob" />
+        <PolarisAutoHiddenInput field="inventoryCount" value={42} />
+        <PolarisAutoSubmit />
+      </PolarisAutoForm>,
+      { wrapper: PolarisMockedProviders }
+    );
+
+    loadMockWidgetCreateMetadata();
+
+    await act(async () => {
+      await user.click(getByRole("button"));
+    });
+
+    const mutation = mockUrqlClient.executeMutation.mock.calls[0][0];
+    const mutationName = mutation.query.definitions[0].name.value;
+    const variables = mutation.variables.widget;
+
+    expect(mutationName).toEqual("createWidget");
+    expect(variables.inventoryCount).toEqual(42);
+    expect(variables.name).toEqual("Bob");
+  });
+
+  it("should override the pre-filled value", async () => {
+    const user = userEvent.setup();
+
+    const { getByRole } = render(
+      <PolarisAutoForm action={api.widget.create} findBy="42">
+        <PolarisAutoHiddenInput field="name" value="Bob" />
+        <PolarisAutoHiddenInput field="inventoryCount" value={9999} />
+        <PolarisAutoSubmit />
+      </PolarisAutoForm>,
+      { wrapper: PolarisMockedProviders }
+    );
+
+    mockUpdateWidgetFindBy();
+
+    await act(async () => {
+      await user.click(getByRole("button"));
+    });
+
+    const mutation = mockUrqlClient.executeMutation.mock.calls[0][0];
+    const mutationName = mutation.query.definitions[0].name.value;
+    const variables = mutation.variables.widget;
+
+    expect(mutationName).toEqual("createWidget");
+    expect(variables.inventoryCount).toEqual(9999);
+    expect(variables.name).toEqual("Bob");
+  });
```

Comment:
Consider a test where you have this
```
        <PolarisAutoHiddenInput field="name" value="ignored because first" />
        <PolarisAutoHiddenInput field="name" value="Bob" />
```
That shows that when there are multiple hidden inputs on the same field with different values, the value that gets sent is the one that is lowest in the DOM